### PR TITLE
Fix for flashing on window creation

### PR DESCRIPTION
**Pull request by rhzk**
Small PR to fix the flashing issue left over from #1037

This should also fix any flashing that might happen when creating a window minimized or maximized.

Also updated setting position and size to use Into

### CODE REVIEW

File: druid-shell/src/window.rs [hunk 1]
```diff
@@ -141,7 +141,7 @@ impl WindowHandle {
     /// Sets the state of the window.
     ///
     /// [`state`]: enum.WindowState.html
-    pub fn set_window_state(&mut self, state: WindowState) {
+    pub fn set_window_state(&self, state: WindowState) {
```
Comment (rjwittams) [inline druid-shell/src/window.rs:144]: It seems odd to do this as a backend may not use interior mutability (especially if we had a completely virtual backend for tests etc).
Does it work on every platform?

Comment (rhzk) [reply druid-shell/src/window.rs:144]: Hmm, having mutable here made the borrow checker complain

Comment (rhzk) [reply druid-shell/src/window.rs]: I will revert this, maybe something to look into in the future when all backends are supported

Comment (rjwittams) [reply druid-shell/src/window.rs]: Anything set_* not taking &mut self or self seems suspect to me... (even if all current back ends do use interior mutability, it logically makes little sense)

**Comment by rhzk** [comment]
There is some edge case that this solution cant handle, I will need to find another solution..

**Pull request closed without merging**